src/Pull.cpp

```diff
@@ -0,0 +1,200 @@
+#include "Pull.h"
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Construct a blocking nanomsg pull (no timeout)
+ */
+Pull::Pull(const std::string& location, const bool shouldConnect) {
+   mProtocolHandler = std::move(std::unique_ptr<NanoProtocol>(
```

Comment:
1) Please use the initializer list initialization instead and there is no reason for the move
2) It's better to use ```std::make_unique``` instead of ```new``` due to location of reference
```
Pull::Pull(const std::string& location, const bool shouldConnect) 
: mProtocolHandler(std::make_unique<NanoProtocol>(location)) 
```